vega_sim/scenario/constant_function_market/scenario.py

```diff
@@ -0,0 +1,459 @@
+        # ]
+
+        # for i_agent in range(1):
+        #     market_agents["random_traders"].append(
+        #         LimitOrderTrader(
+        #             wallet_name=f"RANDOM_TRADERS",
+        #             key_name=(
+        #                 f"LIMIT_{str(i_market).zfill(3)}_AGENT_{str(i_agent).zfill(3)}"
+        #             ),
+        #             market_name=market_name,
+        #             asset_name=asset_name,
+        #             time_in_force_opts={"TIME_IN_FORCE_GTT": 1},
+        #             buy_volume=0.1,
+        #             sell_volume=0.1,
+        #             buy_intensity=1,
+        #             sell_intensity=1,
+        #             submit_bias=1,
+        #             cancel_bias=0,
+        #             duration=120,
+        #             price_process=price_process,
+        #             spread=0,
+        #             mean=-3,
+        #             sigma=0.5,
+        #             tag=(
+        #                 f"MARKET_{str(i_market).zfill(3)}_AGENT_{str(i_agent).zfill(3)}"
+        #             ),
+        #         )
+        #     )
+
+        for _, agent_list in market_agents.items():
+            self.agents.extend(agent_list)
+
+        return {agent.name(): agent for agent in self.agents}
+
+    def configure_environment(
+        self,
+        vega: VegaServiceNull,
+        **kwargs,
+    ) -> MarketEnvironmentWithState:
+        return MarketEnvironmentWithState(
+            agents=list(self.agents.values()),
+            n_steps=self.num_steps,
+            random_agent_ordering=False,
+            transactions_per_block=self.transactions_per_block,
+            vega_service=vega,
+            step_length_seconds=self.step_length_seconds,
+            block_length_seconds=vega.seconds_per_block,
+            pause_every_n_steps=self.pause_every_n_steps,
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--debug", action="store_true")
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.INFO if not args.debug else logging.DEBUG)
+
+    scenario = CFMScenario(
+        num_steps=600,
+        step_length_seconds=10,
+        block_length_seconds=1,
+        transactions_per_block=4096,
+        # pause_every_n_steps=100,
+    )
+
+    with VegaServiceNull(
+        warn_on_raw_data_access=False,
+        run_with_console=False,
+        use_full_vega_wallet=False,
+        retain_log_files=True,
+        launch_graphql=False,
+        seconds_per_block=scenario.block_length_seconds,
+        transactions_per_block=scenario.transactions_per_block,
+    ) as vega:
+        scenario.run_iteration(
+            vega=vega,
+            pause_at_completion=False,
+            log_every_n_steps=10,
+            output_data=True,
+        )
+
+    figs = price_comp_plots()
+    for i, fig in enumerate(figs.values()):
+        fig.savefig(f"./cfm_plots/trading-{i}.jpg")
```

Comment:
I get an error here, manually maxing a `cfm_plots` dir fixed it but may be worth adding a block which creates if it doesn't exist or add the dir it to git.